base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1076 @@
+#define ETDT_USETABTEXTURE  0x00000004
+#define ETDT_ENABLETAB      (ETDT_ENABLE  | ETDT_USETABTEXTURE)
+
+typedef HRESULT (WINAPI * ETDTProc) (HWND, DWORD);
+
+DWORD WINAPI SizeCheck(LPVOID lpParam)
+{
+	HKEY hRegKey = NULL;
+	SETTINGS_INFO SettingsInfo;
+	ZeroMemory(&SettingsInfo, sizeof(SETTINGS_INFO));
+	DWORD dwSize = sizeof(SettingsInfo);
+	
+	if (RegOpenKeyExW(HKEY_CURRENT_USER,
+						L"Software\\ReactOS\\rapps",
+						0,
+						KEY_QUERY_VALUE,
+						&hRegKey) != ERROR_SUCCESS)
+	{
+		DPRINT("RegOpenKeyExW(): Failed to open a registry key!\n");
+	}
+	
+	HWND HProgressBar = NULL;
+	WCHAR TargetedDir[MAX_PATH] = { 0 };
+	
+	HProgressBar = GetDlgItem(dv.hwndDlg, IDC_PROGRESS_1);
+	
+	if(bv.SysDrive == TRUE)
+	{
+		SHGetFolderPathW(NULL, CSIDL_WINDOWS, NULL, SHGFP_TYPE_CURRENT, TargetedDir);
+		StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\Temp");
+		if (PathIsDirectoryW(TargetedDir))
+		{
+			sz.TempASize = DirSizeFunc(TargetedDir);
+			if (sz.TempASize == 1)
+			{
+				return FALSE;
+			}
+			StringCbCopyW(wcv.TempDir, _countof(wcv.TempDir), TargetedDir);
+			ZeroMemory(&TargetedDir, sizeof(TargetedDir));
+		}
+		SHGetFolderPathW(NULL, CSIDL_PROFILE, NULL, SHGFP_TYPE_CURRENT, TargetedDir);
+		StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\Local Settings\\Temp");
+		if (PathIsDirectoryW(TargetedDir))
+		{
+			sz.TempBSize = DirSizeFunc(TargetedDir);
+			if (sz.TempBSize == 1)
+			{
+				return FALSE;
+			}
+			SetDlgItemTextW(dv.hwndDlg, IDC_STATIC_INFO, L"Temporary Files");
+			StringCbCopyW(wcv.AltTempDir, _countof(wcv.AltTempDir), TargetedDir);
+			ZeroMemory(&TargetedDir, sizeof(TargetedDir));
+		}
+
+		SendMessageW(HProgressBar, PBM_SETPOS, 25, 0);
+		Sleep(150);
+	}
+
+	if(FindRecycleBin(wcv.DriveLetter))
+	{
+		StringCchPrintfW(TargetedDir, _countof(TargetedDir), L"%s\\%s", wcv.DriveLetter, wcv.RecycleBin);
+		sz.RecycleBinSize = DirSizeFunc(TargetedDir);
+		if (sz.RecycleBinSize == 1)
+		{
+			return FALSE;
+		}
+		StringCbCopyW(wcv.RecycleBinDir, _countof(wcv.RecycleBinDir), TargetedDir);
+		SetDlgItemTextW(dv.hwndDlg, IDC_STATIC_INFO, L"Recycled Files");
+		ZeroMemory(&TargetedDir, sizeof(TargetedDir));
+		SendMessageW(HProgressBar, PBM_SETPOS, 50, 0);
+		Sleep(150);
+	}
+
+	
+	for(int i = 0; i < 10; i++)
+	{
+		StringCchPrintfW(TargetedDir, _countof(TargetedDir), L"%s\\FOUND.%.3i", wcv.DriveLetter, i);
+		if (PathIsDirectoryW(TargetedDir))
+		{
+			sz.ChkDskSize += DirSizeFunc(TargetedDir);
+			if (sz.ChkDskSize == 1)
+			{
+				return FALSE;
+			}
+			SetDlgItemTextW(dv.hwndDlg, IDC_STATIC_INFO, L"Old ChkDsk Files");
+			ZeroMemory(&TargetedDir, sizeof(TargetedDir));
+			SendMessageW(HProgressBar, PBM_SETPOS, 75, 0);
+		}
+		
+		else
+			break;
+	}
+	
+	/* Checking if there is a custom path for RAPPS */
+	if(RegQueryValueExW(hRegKey, L"Settings", 0, NULL, (LPBYTE)&SettingsInfo, &dwSize) == ERROR_SUCCESS)
+		StringCbCopyW(TargetedDir, _countof(TargetedDir), SettingsInfo.szDownloadDir);
```

Comment:
This is a hack and will break with every update of rapps.

Reply:
@learn-more So notify that this is a hack?